### ISSUE TITLE
all place pages throwing archetypes error

### ISSUE THREAD

**Issue by paregorios** [isawnyu/pleiades-gazetteer#189]
For example:

http://pleiades.stoa.org/places/74935822

```
root@isaw1:/srv/python27-apps/pleiades4/var/log# grep -A 120 '1468490498.90.419089611281' instance5.log
2016-07-14T06:01:38 ERROR Zope.SiteErrorLog 1468490498.90.419089611281 http://pleiades.stoa.org/places/74935822/base_view
Traceback (innermost last):
  Module ZPublisher.Publish, line 138, in publish
  Module ZPublisher.mapply, line 77, in mapply
  Module ZPublisher.Publish, line 48, in call_object
  Module Shared.DC.Scripts.Bindings, line 322, in __call__
  Module Shared.DC.Scripts.Bindings, line 359, in _bindAndExec
  Module Products.CMFCore.FSPageTemplate, line 237, in _exec
  Module Products.CMFCore.FSPageTemplate, line 177, in pt_render
  Module Products.PageTemplates.PageTemplate, line 87, in pt_render
  Module zope.pagetemplate.pagetemplate, line 132, in pt_render
   - Warning: Macro expansion failed
   - Warning: <type 'exceptions.KeyError'>: 'macro'
  Module zope.pagetemplate.pagetemplate, line 240, in __call__
  Module collective.newrelic.patches.talinterpreter, line 17, in monkeypatch
  Module newrelic.api.function_trace, line 110, in literal_wrapper
  Module zope.tal.talinterpreter, line 271, in __call__
  Module zope.tal.talinterpreter, line 343, in interpret
  Module zope.tal.talinterpreter, line 888, in do_useMacro
  Module zope.tal.talinterpreter, line 343, in interpret
  Module zope.tal.talinterpreter, line 533, in do_optTag_tal
  Module zope.tal.talinterpreter, line 518, in do_optTag
  Module zope.tal.talinterpreter, line 513, in no_tag
  Module zope.tal.talinterpreter, line 343, in interpret
  Module zope.tal.talinterpreter, line 954, in do_defineSlot
  Module zope.tal.talinterpreter, line 343, in interpret
  Module zope.tal.talinterpreter, line 533, in do_optTag_tal
  Module zope.tal.talinterpreter, line 518, in do_optTag
  Module zope.tal.talinterpreter, line 513, in no_tag
  Module zope.tal.talinterpreter, line 343, in interpret
  Module zope.tal.talinterpreter, line 858, in do_defineMacro
  Module zope.tal.talinterpreter, line 343, in interpret
  Module zope.tal.talinterpreter, line 954, in do_defineSlot
  Module zope.tal.talinterpreter, line 343, in interpret
  Module zope.tal.talinterpreter, line 533, in do_optTag_tal
  Module zope.tal.talinterpreter, line 518, in do_optTag
  Module zope.tal.talinterpreter, line 513, in no_tag
  Module zope.tal.talinterpreter, line 343, in interpret
  Module zope.tal.talinterpreter, line 954, in do_defineSlot
  Module zope.tal.talinterpreter, line 343, in interpret
  Module zope.tal.talinterpreter, line 533, in do_optTag_tal
  Module zope.tal.talinterpreter, line 518, in do_optTag
  Module zope.tal.talinterpreter, line 513, in no_tag
  Module zope.tal.talinterpreter, line 343, in interpret
  Module zope.tal.talinterpreter, line 946, in do_defineSlot
  Module zope.tal.talinterpreter, line 343, in interpret
  Module zope.tal.talinterpreter, line 533, in do_optTag_tal
  Module zope.tal.talinterpreter, line 518, in do_optTag
  Module zope.tal.talinterpreter, line 513, in no_tag
  Module zope.tal.talinterpreter, line 343, in interpret
  Module zope.tal.talinterpreter, line 858, in do_defineMacro
  Module zope.tal.talinterpreter, line 343, in interpret
  Module zope.tal.talinterpreter, line 533, in do_optTag_tal
  Module zope.tal.talinterpreter, line 518, in do_optTag
  Module zope.tal.talinterpreter, line 513, in no_tag
  Module zope.tal.talinterpreter, line 343, in interpret
  Module zope.tal.talinterpreter, line 888, in do_useMacro
  Module zope.tal.talinterpreter, line 343, in interpret
  Module zope.tal.talinterpreter, line 533, in do_optTag_tal
  Module zope.tal.talinterpreter, line 518, in do_optTag
  Module zope.tal.talinterpreter, line 513, in no_tag
  Module zope.tal.talinterpreter, line 343, in interpret
  Module zope.tal.talinterpreter, line 821, in do_loop_tal
  Module zope.tal.talinterpreter, line 343, in interpret
  Module zope.tal.talinterpreter, line 533, in do_optTag_tal
  Module zope.tal.talinterpreter, line 518, in do_optTag
  Module zope.tal.talinterpreter, line 513, in no_tag
  Module zope.tal.talinterpreter, line 343, in interpret
  Module zope.tal.talinterpreter, line 852, in do_condition
  Module zope.tal.talinterpreter, line 343, in interpret
  Module zope.tal.talinterpreter, line 533, in do_optTag_tal
  Module zope.tal.talinterpreter, line 518, in do_optTag
  Module zope.tal.talinterpreter, line 513, in no_tag
  Module zope.tal.talinterpreter, line 343, in interpret
  Module zope.tal.talinterpreter, line 852, in do_condition
  Module zope.tal.talinterpreter, line 343, in interpret
  Module zope.tal.talinterpreter, line 533, in do_optTag_tal
  Module zope.tal.talinterpreter, line 518, in do_optTag
  Module zope.tal.talinterpreter, line 513, in no_tag
  Module zope.tal.talinterpreter, line 343, in interpret
  Module zope.tal.talinterpreter, line 888, in do_useMacro
  Module zope.tal.talinterpreter, line 343, in interpret
  Module zope.tal.talinterpreter, line 888, in do_useMacro
  Module zope.tal.talinterpreter, line 343, in interpret
  Module zope.tal.talinterpreter, line 583, in do_setLocal_tal
  Module zope.tales.tales, line 696, in evaluate
   - URL: file:/srv/python27-apps/pleiades4/eggs/Products.Archetypes-1.9.10-py2.7.egg/Products/Archetypes/skins/archetypes/widgets/field.pt
   - Line 12, Column 4
   - Expression: <PythonExpr context.widget(field.getName(), mode=mode, use_label=1)>
   - Names:
      {'container': <Place at /plone/places/74935822>,
       'context': <Place at /plone/places/74935822>,
       'default': <object object at 0x7f6f27b6dbb0>,
       'here': <Place at /plone/places/74935822>,
       'loop': {u'field': <Products.PageTemplates.Expressions.PathIterator object at 0x7f6ec685fed0>},
       'nothing': None,
       'options': {'args': ()},
       'repeat': <Products.PageTemplates.Expressions.SafeMapping object at 0x7f6ec685d680>,
       'request': <HTTPRequest, URL=http://pleiades.stoa.org/places/74935822/base_view>,
       'root': <Application at >,
       'template': <FSPageTemplate at /plone/places/74935822/base_view>,
       'traverse_subpath': [],
       'user': <SpecialUser 'Anonymous User'>}
  Module Products.PageTemplates.ZRPythonExpr, line 48, in __call__
   - __traceback_info__: context.widget(field.getName(), mode=mode, use_label=1)
  Module PythonExpr, line 1, in <expression>
  Module Products.Archetypes.BaseObject, line 287, in widget
  Module Products.Archetypes.Renderer, line 26, in render
  Module Products.Archetypes.generator.widget, line 147, in __call__
AttributeError: Macro filtered_inandout does not exist for <Place at 74935822>
```

**Comment by paregorios** [comment]
Possibly related? 

```
Hello New Relic Account Owners,

We are making a number of changes to the “metric names” endpoints in our API this month.

You are receiving this email because you are the owner of a New Relic account, or owner of a master account (with sub-accounts) that have used one or more of the affected API endpoints in the past few weeks. 

While these are not breaking changes, you may have relied on previously existing API behavior. 

This Thursday, July 14th, New Relic will make two changes to the “metric names” endpoints in our API:

The metric names in the response will be ordered differently. The sort order is not guaranteed and we will not offer a guaranteed ordering.
Response pages will contain up to 1,000 results, rather than the previous maximum of 100.
Later this month, another change will be deployed to the “metric names” endpoint:

A new pagination model that makes getting your metric names more efficient for both you and New Relic. This pagination method will soon appear in the API Explorer, and you may choose to use either the new cursor-based method, or the old page number method.
Note: There may be more changes occurring this month for “metric names” endpoints beyond the three listed above. Please review https://discuss.newrelic.com/t/upcoming-changes-to-all-metric-names-api-endpoints/38993 and “watch” that topic to be notified as more information becomes available.

If you have questions, please post to that discussion thread, or reply to this email.

Please forward this email to your colleagues that may work on integrations with New Relic's API. 

Thanks, 
Cooper Marcus
Senior Product Manager - APIs and Shared Services 
cooper@newrelic.com
```

**Comment by paregorios** [comment]
regular pages (blog, downloads, etc.) seem to be ok

**Comment by paregorios** [comment]
This problem does not seem to be occurring on staging.

**Comment by alecpm** [comment]
This appears to be related to https://github.com/isawnyu/PleiadesEntity/commit/21ce0485be335ad203346a897e3b7bdea2479eea

Is there a missing dependency or version mismatch between staging and production?

**Comment by davisagli** [comment]
Oh dear, when I deployed the fix for #186 last night I missed checking in a new file. Deploying the fix now.

**Comment by davisagli** [comment]
Fixed. My apologies, I should have tested after deploying.
